rskj-core/src/main/java/co/rsk/trie/IterationElement.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2022 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package co.rsk.trie;
+
+public class IterationElement {
+
+        private final TrieKeySlice nodeKey;
+        private final Trie node;
+
+        public IterationElement(final TrieKeySlice nodeKey, final Trie node) {
+            this.nodeKey = nodeKey;
+            this.node = node;
+        }
+
+        public Trie getNode() {
+            return node;
+        }
+
+        public final TrieKeySlice getNodeKey() {
+            return nodeKey;
+        }
+
+        public String toString() {
+            byte[] encodedFullKey = nodeKey.expand();
```

Comment:
Was the change from `encode` to `expand` intentional?

Reply:
Encode meaning depends on the type of encoding you're using. Everything is "encoding". 
In the next PR I plan to replace the TrieKeySlice of another class which uses different encoding, the word "encode" will be even more confusing. The word "expand" means encode it in a way where 1s and 0s are expanded one element in each byte.

Reply:
I think the implementation of "expand" is exactly equal to the previous implementation of "encode".  So logs won't change. I can check this.

Reply:
I think the implementation is a little bit different, since the old `encode` method calls to `PathEncoder.encode(...`. Did we check the thing regarding the logs changing yet?

Reply:
@SergioDemianLerner , any updates on this?

Reply:
@SergioDemianLerner , it seems that this is the only pending thing for this PR)

` nodeKey.expand()`'s implementation is following:
```
    // Returns the key slice in expanded form (byte array of ones and zeros)
    public byte[] expand() {
        return Arrays.copyOfRange(expandedKey, offset, limit);
    }
```
and the impl of the `encode()` method:
```
    public byte[] encode() {
        // TODO(mc) avoid copying by passing the indices to PathEncoder.encode
        return PathEncoder.encode(Arrays.copyOfRange(expandedKey, offset, limit));
    }
```

also it looks like that `PathEncoder.encode(bytes) != bytes`, which means that `IterationElement.toString()` would return a bit different strings, and as a result, it'd have an impact on values being logged:
```
public String toString() {
            byte[] encodedFullKey = nodeKey.expand();
```

is this intentional? shouldn't we use an old format that's being returned by `nodeKey.encode()`?

Reply:
Now I see what you mean. The change was intentional to let me more easily debug. This is because I'm using different Trie key classes and I want all of them to log in the same simple-to-understand format (not the internal format). But I'm ok to revert it if you are use to reading logs in a specific format. But I guess nobody (except me) cares what was the previous logging format.
So I leave that question here: do you care of the new logging format ?

Reply:
yeah, agree. Probably no one else besides you has been using this format.